### PR TITLE
RavenDB-7099 and RavenDB-7097

### PR DESCRIPTION
**Pull request by arekpalinski**

### CODE REVIEW

File: src/Voron/StorageEnvironmentOptions.cs [hunk 1]
```diff
@@ -471,6 +473,20 @@ protected override void Disposing()
                     if (journal.Value.IsValueCreated)
                         journal.Value.Value.Dispose();
                 }
+
+                lock (_journalsForReuse)
+                {
+                    foreach (var reusableFile in _journalsForReuse.Values)
```
Comment (ayende) [inline src/Voron/StorageEnvironmentOptions.cs:479]: This make me wonder if we wants this in this manner. Imagine a case where you are shutting down a busy server, you don't want to lose these files. 

Wouldn't it be better to use it with a flag that will prevent pending recycle  in `RecoverDatabase`?

**Review by ayende — CHANGES_REQUESTED** [review]

**Comment by ayende** [comment]
I'm also not seeing where in 	d11877f you prevented the creation?

**Comment by arekpalinski** [comment]
Creation prevented here:
https://github.com/ravendb/ravendb/pull/2407/files#diff-1b04de6ee2cc09d658cbd72076e98a51L488

by removing `TryStoreJournalForReuse` from `TryDeleteJournal` which is called only during `RecoverDatabase`

File: src/Voron/StorageEnvironmentOptions.cs (hunk 1 above)
Comment (arekpalinski) [reply src/Voron/StorageEnvironmentOptions.cs:479]: I don't follow what a flag to use you mean

**Comment by ayende** [comment]
The `TryDeleteJournal` change was non obvious, and slipped by.

File: src/Voron/StorageEnvironmentOptions.cs (hunk 1 above)
Comment (ayende) [reply src/Voron/StorageEnvironmentOptions.cs:479]: You are deleting the recycled journals on shutdown. But is that something we want? 
If we restart, we will lose them.

On the other hand, that means that we don't take extra space for databases that have been shutdown...

Comment (arekpalinski) [reply src/Voron/StorageEnvironmentOptions.cs:479]: I'm for the approach that the recycle areas (for scratches and journals) exist only when a db is live. Previously we were not consistent here - recycled scratches were deleted while scratches were not.

Comment (arekpalinski) [reply src/Voron/StorageEnvironmentOptions.cs:479]: *while recycled journals

**Pull request merged**